Code/Framework/AzCore/Tests/DOM/DomJsonBenchmarks.cpp

```diff
@@ -0,0 +1,184 @@
+            };
+
+            auto createEntry = [&](int n) -> rapidjson::Value
+            {
+                rapidjson::Value entry(rapidjson::kObjectType);
+                entry.AddMember("string", createString(n), document.GetAllocator());
+                entry.AddMember("int", rapidjson::Value(n), document.GetAllocator());
+                entry.AddMember("double", rapidjson::Value(static_cast<double>(n) * 0.5), document.GetAllocator());
+                entry.AddMember("bool", rapidjson::Value(n % 2 == 0), document.GetAllocator());
+                entry.AddMember("null", rapidjson::Value(rapidjson::kNullType), document.GetAllocator());
+                return entry;
+            };
+
+            auto createArray = [&]() -> rapidjson::Value
+            {
+                rapidjson::Value array;
+                array.SetArray();
+                for (int i = 0; i < entryCount; ++i)
+                {
+                    array.PushBack(createEntry(i), document.GetAllocator());
+                }
+                return array;
+            };
+
+            auto createObject = [&]() -> rapidjson::Value
+            {
+                rapidjson::Value object;
+                object.SetObject();
+                for (int i = 0; i < entryCount; ++i)
+                {
+                    buffer = AZStd::string::format("Key%i", i);
+                    rapidjson::Value key;
+                    key.SetString(buffer.data(), static_cast<rapidjson::SizeType>(buffer.length()), document.GetAllocator());
+                    object.AddMember(key.Move(), createArray(), document.GetAllocator());
+                }
+                return object;
+            };
+
+            document.SetObject();
+            document.AddMember("entries", createObject(), document.GetAllocator());
+
+            AZStd::string serializedJson;
+            auto result = AZ::JsonSerializationUtils::WriteJsonString(document, serializedJson);
+            AZ_Assert(result.IsSuccess(), "Failed to serialize generated JSON");
+            return serializedJson;
+        }
+    };
+
+// Helper macro for registering JSON benchmarks
+#define BENCHMARK_REGISTER_JSON(BaseClass, Method)                                                                                         \
+    BENCHMARK_REGISTER_F(BaseClass, Method)                                                                                                \
+        ->Args({ 10, 5 })                                                                                                                  \
+        ->Args({ 10, 500 })                                                                                                                \
+        ->Args({ 100, 5 })                                                                                                                 \
+        ->Args({ 100, 500 })                                                                                                               \
```

Comment:
Perhaps extends these with a few additional values because it's possible for tools to generate documents with thousands of values.

Reply:
fwiw the first parameter is O(n**2) which I can call out better. I've tried with 100K-1M values instead of 10K but it didn't lead to any significantly different throughput results so I omitted it to save some time - can up the size if there's some threshold you're interested in seeing.

Reply:
Just interested in providing enough data points to have a reliable complexity measure.